test/conformance/kernel/urKernelGetGroupInfo.cpp

```diff
@@ -3,56 +3,156 @@
-                      UR_KERNEL_GROUP_INFO_COMPILE_WORK_GROUP_SIZE,
-                      UR_KERNEL_GROUP_INFO_LOCAL_MEM_SIZE,
-                      UR_KERNEL_GROUP_INFO_PREFERRED_WORK_GROUP_SIZE_MULTIPLE,
-                      UR_KERNEL_GROUP_INFO_PRIVATE_MEM_SIZE,
-                      UR_KERNEL_GROUP_INFO_COMPILE_MAX_WORK_GROUP_SIZE,
-                      UR_KERNEL_GROUP_INFO_COMPILE_MAX_LINEAR_WORK_GROUP_SIZE),
-    uur::deviceTestWithParamPrinter<ur_kernel_group_info_t>);
-
-struct urKernelGetGroupInfoSingleTest : uur::urKernelTest {
+struct urKernelGetGroupInfoFixedWorkGroupSizeTest : uur::urKernelTest {
     void SetUp() override {
+        program_name = "fixed_wg_size";
         UUR_RETURN_ON_FATAL_FAILURE(urKernelTest::SetUp());
     }
+
+    // This value correlates to work_group_size<8, 4, 2> in fixed_wg_size.cpp.
+    // In SYCL, the right-most dimension varies the fastest in linearization.
+    // In UR, this is on the left, so we reverse the order of these values.
+    std::array<size_t, 3> work_group_size{2, 4, 8};
 };
-UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urKernelGetGroupInfoSingleTest);
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urKernelGetGroupInfoFixedWorkGroupSizeTest);
 
-struct urKernelGetGroupInfoWgSizeTest : uur::urKernelTest {
+struct urKernelGetGroupInfoMaxWorkGroupSizeTest : uur::urKernelTest {
     void SetUp() override {
-        program_name = "fixed_wg_size";
+        program_name = "max_wg_size";
         UUR_RETURN_ON_FATAL_FAILURE(urKernelTest::SetUp());
     }
 
-    // This must match the size in fixed_wg_size.cpp
-    std::array<size_t, 3> wg_size{4, 4, 4};
+    // These values correlate to max_work_group_size<6, 5, 4> and
+    // max_linear_work_group_size<120> in max_wg_size.cpp.
+    // In SYCL, the right-most dimension varies the fastest in linearization.
+    // In UR, this is on the left, so we reverse the order of these values.
+    std::array<size_t, 3> max_work_group_size{2, 4, 8};
+    size_t max_linear_work_group_size{64};
 };
-UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urKernelGetGroupInfoWgSizeTest);
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urKernelGetGroupInfoMaxWorkGroupSizeTest);
 
-TEST_P(urKernelGetGroupInfoTest, Success) {
-    auto property_name = getParam();
-    size_t property_size = 0;
-    std::vector<char> property_value;
-    auto result = urKernelGetGroupInfo(kernel, device, property_name, 0,
-                                       nullptr, &property_size);
-    if (result == UR_RESULT_SUCCESS) {
-        property_value.resize(property_size);
-        ASSERT_SUCCESS(urKernelGetGroupInfo(kernel, device, property_name,
-                                            property_size,
-                                            property_value.data(), nullptr));
-    } else {
-        ASSERT_EQ_RESULT(result, UR_RESULT_ERROR_UNSUPPORTED_ENUMERATION);
+struct urKernelGetGroupInfoTest : uur::urKernelTest {
+    void SetUp() override {
+        UUR_RETURN_ON_FATAL_FAILURE(urKernelTest::SetUp());
```

Comment:
is this missing a `program_name = `?

Reply:
No. This is the generic test case for which the values returned aren't checked. Before these changes it was the parameterized `urKernelGetGroupInfoTest` and the `urKernelGetGroupInfoSingleTest`, neither of which had a specified `program_name`.

Reply:
alright cool, think it could be reduced to just `using urKernelGetGroupInfoTest = uur::urKernelTest;` in that case

Reply:
Ah yes. I thought it would fail on `Build()` but appears to work - thanks! Done.